### PR TITLE
New modules for the administration of IBM DataPower Gateway with Ansible

### PR DESCRIPTION
**Pull request by dgraum**
##### SUMMARY

This is the beginning of the implementation of a solution to manage the IBM DataPower Gateway (https://www.ibm.com/products/datapower-gateway) with Ansible. I suggest that this development be stored in a new "modules / appliance" directory that could serve as a collaboration space for the technologies produced by this and other technology companies.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

- idg_domain

##### ANSIBLE VERSION

```
ansible 2.6.0
  config file = /home/dgrau/Documents/Projects/idg_ansible/datapower-ansible/ansible.cfg
  configured module search path = [u'/home/dgrau/Documents/Projects/idg_ansible/datapower-ansible/library/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.15 (default, May 15 2018, 15:37:31) [GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
```

##### ADDITIONAL INFORMATION

The combination of both technologies promises many advantages for the users of these IBM devices and their new proposals in the cloud.

### CODE REVIEW

File: lib/ansible/module_utils/appliance/ibm/idg_common.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    #############################
+
+    IMMUTABLE_MESSAGE = 'The current state is consistent with the desired configuration.'
+    COMPLETED_MESSAGE = 'Completed.'
+    CHECK_MODE_MESSAGE = 'Change was only simulated, due to enabling verification mode.'
+    UNCONTROLLED_EXCEPTION = 'Unknown exception.'
+
+    # Connection agreements
+    BASIC_HEADERS = {"Content-Type": "application/json"}
+    ANSIBLE_VERSION = '2.7'
```
Comment (gundalow) [inline lib/ansible/module_utils/appliance/ibm/idg_common.py:57]: Their should be a variable you can read in, rather than hardcoding this

**Review by gundalow — CHANGES_REQUESTED** [review]
Hi,
Thank you for this PR.
I wonder if we can find a better directory for this than `appliance/ibm`. Do any of the existing directory suite this?

What exactly is IBM DataPower Gateway?

File: lib/ansible/module_utils/appliance/ibm/idg_common.py (hunk 1 above)
Comment (dgraum) [reply lib/ansible/module_utils/appliance/ibm/idg_common.py:57]: Hi John!

IBM DataPower® Gateway serves as the enterprise-grade, field-proven application integration for SOA and REST and security-rich API gateway.
This technology has maintained steady growth and is seen as an important component of IBM's cloud strategy.

[https://www.ibm.com/products/datapower-gateway](https://www.ibm.com/products/datapower-gateway)

I also do not feel very comfortable with the current route, but I did not find a position that I could reuse, maybe something like: gateway, api, ...?

Thanks to the corrections, we will work to ensure that the appropriate corrections are made. 

Thank you for your attention.

**Comment by Akasurde** [comment]
Hi @dgraum, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.general<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>

**Pull request closed without merging**